app/domain/authentication/strategy.rb

```diff
@@ -77,6 +77,24 @@ def conjur_token(input)
       raise e
     end
 
+    def conjur_token_oidc(input)
+      authenticator = authenticators[input.authenticator_name]
+
+      validate_authenticator_exists(input, authenticator)
+
+      # this method also injects the username to input
+      validate_credentials(input, authenticator)
+
+      validate_security(input)
+      validate_origin(input)
+
+      audit_success(input)
+      new_token(input)
+    rescue => e
+      audit_failure(input, e)
+      raise e
+    end
+
```

Comment:
This method is problematic in itself as well as for the ripple effects it causes in the `Authenticator` itself (discussed in other comments), but I suspect it will go away naturally when we implement the new design, in which retrieving the token will be its own API route.  That is, when the `authenticate` route becomes simply exchanging the session cookie containing the encrypted id token for a conjur token, then it will fit within the original `Strategy` object, without modification.

I think the additional route of the new design may naturally clean up some of the issues I mentioned in other comments as well.

Reply:
let's get back to this after the new design is implemented and see if it's still a problem